src/pybind/mgr/telemetry/module.py

```diff
@@ -516,12 +528,17 @@ def gather_perf_counters(self) -> Dict[str, dict]:
         result: Dict[str, dict] = defaultdict(lambda: defaultdict(
             lambda: defaultdict(lambda: defaultdict(int))))
 
-        # Condense metrics among like daemons (i.e. 'osd' = 'osd.0' +
-        # 'osd.1' + 'osd.2'), and update the 'result' dict.
         for daemon in all_perf_counters:
-            daemon_type = daemon[0:3] # i.e. 'mds', 'osd', 'rgw'
-            for collection in all_perf_counters[daemon]:
 
+            # Calculate num combined daemon types if in aggregated mode
+            if mode == 'aggregated':
+                daemon_type = daemon[0:3] # i.e. 'mds', 'osd', 'rgw'
+                if 'num_combined_daemons' not in result[daemon_type]:
+                    result[daemon_type]['num_combined_daemons'] = 1
+                else:
+                    result[daemon_type]['num_combined_daemons'] += 1
```

Comment:
I calculate the number of combined daemons for perf counters here. I was hoping for a more elegant way of doing it (similar to how we initialize the `count` and `value` fields as `int` in the `result` dict). But since there doesn't seem to be a way to initialize multiple default values, I went with this tried-and-true method.

I could alternatively do something like `result[daemon_type]['num_combined_daemons'] = result[daemon_type].get('num_combined_daemons', 0) + 1`, but I felt that the other solution was more readable.

Reply:
this works